tables/tabulate-molecular-alterations.R

```diff
@@ -0,0 +1,264 @@
+fusion_df      <-  dplyr::filter(fusion_df, Sample %in% bs_ids) 
+cnv_df         <-  dplyr::filter(cnv_df, biospecimen_id %in% bs_ids) 
+
+# What are all the bs ids that end up being considered? 
+# This variable can be used when joining things back up
+final_bs_ids <- length(unique(
+  c(
+    maf_df$Tumor_Sample_Barcode,
+    fusion_df$Sample,
+    cnv_df$biospecimen_id
+)))
+
+
+
+#### Prepare maf, cnv, fusion data ------------------
+
+maf_df <- prepare_maf(maf_df)
+fusion_df <- prepare_fusion(fusion_df)
+cnv_df <- prepare_cnv(cnv_df)
+
+
+### Combine alterations and filter to GOI ---------------------
+
+# We specifically are interested in showing the Figure 2 (and S3B which has same genes as 2D) alterations but for all tumors.
+# The gene_list below contains the top GOIs identified in `scripts/figures/fig2_figS3-oncoprint-landscape.R`, sorted and de-duplicated
+# For further reference, see Figure 2:
+# https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/aa929753fca0019294571ec813e6ae7224b1d8b8/figures/pngs/figure2.png
+gene_list <- c("ACVR1", "APC", "ATM", "ATRX", "BCOR", "BRAF", 
+               "C11orf95", "CDK6", "CTDNEP1", "CTNNB1", "DDX3X", 
+               "EGFR", "ERBB4", "EWSR1", "FGFR1", "FGFR2", "FOXO3", 
+               "H3F3A", "IGF1R", "KDM6A", "KIAA1549", "KIT", 
+               "KMT2C", "KMT2D", "KRAS", "KSR2", "MAML2", "MAMLD1", 
+               "MET", "MIR512-2", "MYB", "MYCN", "NF1", "NF2", 
+               "NTRK2", "NTRK3", "PDGFRA", "PIK3CA", "PIK3R1", 
+               "PPM1D", "PRKAR1A", "PTCH1", "PTEN", "PTPN11", 
+               "QKI", "RAF1", "RELA", "ROS1", "SETD2", "SMARCA4", 
+               "TACC1", "TCF4", "TERT", "TP53", "YAP1", "ZIC1")
+
+select_cols <- c("Tumor_Sample_Barcode", "Hugo_Symbol", "Variant_Classification")
+
+
+# Combine all alterations:
+alterations_df <- list(maf_df, fusion_df, cnv_df) %>%
+  # keeping it interesting
+  purrr::map(
+    dplyr::select,
+    select_cols
+  ) %>%
+  dplyr::bind_rows() %>%
+  # rename Tumor_Sample_Barcode to sample_id
+  dplyr::rename(sample_id = Tumor_Sample_Barcode) %>%
+  # filter to only genes in gene_list
+  dplyr::filter(Hugo_Symbol %in% gene_list) %>%
+  # some wrangling to help with dev:
+  dplyr::arrange(Hugo_Symbol) %>%
+  tibble::as_tibble()
+
+# Ensure we still have all the genes at least _somewhere_ in the data frame
+if (!(length(unique(alterations_df$Hugo_Symbol)) == length(gene_list))) {
+  stop("Some genes are missing from the alteration df.")
+}
+
+# Convert to presence/absence of _all combinations_ of genes and variants for each sample_id
+alterations_presence_df <- alterations_df %>%
+  # If the row is here already, then the alteration is present
+  dplyr::mutate(present = "Y") %>% 
+  # Fill in all combinatins of gene/variant for each sample, 
+  #  filling in "N" (no) for present for any new rows 
+  tidyr::complete(sample_id, Hugo_Symbol, Variant_Classification, 
+                  fill = list(present = "N")) %>%
```

Comment:
Any reason not to use T/F here?
```suggestion
  dplyr::mutate(present = TRUE) %>% 
  # Fill in all combinatins of gene/variant for each sample, 
  #  filling in FALSE for present for any new rows 
  tidyr::complete(sample_id, Hugo_Symbol, Variant_Classification, 
                  fill = list(present = FALSE)) %>%
```